README.md

```diff
@@ -29,22 +29,24 @@ length limit is based on a survey of the defaults for various JSON parsers.
 extension which MUST reference all the parameters.
 * Type: Data type for the parameter. In order to allow validation and limit the scope for implementation specific variations,
 parameters are typed.
-  * integer - may be signed or not as specified in the range parameter.
+  * integer - may be signed or not as specified in the range property.
   * string - aligned with JSON strings, these should be UTF-8 encoded and avoid control characters.  
   * enumerated - one of an ordered set of labels which MUST conform to the same limitations as parameter names.  
-  * boolean - may have the values false or true (lower case and unquoted as in JSON)
-  * For integer parameters the range specifies minimum and maximum values, separated by a comma, which MUST be integers themselves.
-  * For string parameters, the range specifies the maximum length of the string as an integer number of characters, not bytes. Again, based on a survey of parsers, try to keep strings shorter than 4095 characters.
+  * boolean - MUST have the value *false* or *true* (lower case and unquoted as in JSON)
+* Range: Further qualifies the valid values for a paramater depending on its type.
+  * For integer parameters, the range specifies minimum and maximum values, separated by a comma, which MUST be integers themselves.
+  * For string parameters, the range specifies the maximum length of the string as an integer number of characters, not bytes. Again, based on a survey of parsers, strings SHOULD be shorter than 4095 characters.
   * For enumerated parameters, the range is a comma separated ordered list of labels that are valid for the parameter. Enumerated parameters are case sensitive and MUST always be quoted, so they are JSON strings. 
-* Default: Default value for parameter, which MUST be consistent with the range limitations. If this is left blank then the parameter is mandatory 
+* Default: Default value for parameter, which MUST be consistent with the range limitations. If this is left blank then the parameter is mandatory (if parameters are defined)  
 
 ## Referencing Parameters
 
-Wherever a parameterised extension is referenced, include any parameters in an accompanying JSON file. If using an extensions directory, the JSON file MUST be named for the extension and included in the directory. For example, the example extension above would have an accompanying file *0000-example-extension.json* which might contain:
+Wherever a parameterised extension is referenced, any parameters MAY be included in an accompanying JSON file. If using an extensions directory, the JSON file MUST be named for the extension and included in the directory. For example, the example extension above would have an accompanying file *0000-example-extension.json* which might contain:
```

Comment:
I'm a little confused by this wording:

> If using an extensions directory, the JSON file MUST be named for the extension and included in the directory.

1. Is it acceptable to put these JSON files anywhere other than in the `extensions` directory?
1. If it is put inside the `extensions` directory, does it matter where? For example, `extensions/0000-example-extension.json` or `extensions/0000-example-extension/0000-example-extension.json`.

Reply:
1. At the moment, there isn't another way of referencing extensions other than the extension directory so no.
2. Interesting point, the intent was the extension directory and not a subdirectory.

Reply:
Clarified that it needs to be in the extension directory and not a subdirectory. Also, ocfl_layout.json is another place where a parameter sidecar file can occur.

Reply:
@neilsjefferies Thanks for the updates!

The [spec currently states](https://ocfl.io/draft/spec/#object-extensions) that extensions should use a subdirectory of `extensions` "named according to a registered extension name". I don't necessary have a problem with the `json` files being directly under `extensions`. In fact, for extensions that do not have additional files, this makes a lot of sense. But, it is a little more awkward for extensions that have both configuration and a sub directory. For example:

```
extensions/
 - 0000-example-extension.json
 - 0000-example-extension/
    - ... some files ...
```

In those cases, it would be neater if the json file was under the extension directory. Perhaps, the json file should be at the root of `extensions` when the extension has no other files and under a subdirectory if it does? Not a huge deal either way.

Reply:
Actually, I agree! I intended that the extension parameters should be in the "root directory of the extension" not extensions since it shouldn't contain files as per the spec. Let me reword that...

Reply:
Sounds good to me too